Xamarin.Forms.Core/Cells/Cell.cs

```diff
@@ -18,6 +19,24 @@ public abstract class Cell : Element, ICellController
 
 		bool _nextCallToForceUpdateSizeQueued;
 
+		EffectiveFlowDirection _effectiveFlowDirection = EffectiveFlowDirection.LeftToRight | EffectiveFlowDirection.Implicit;
+		EffectiveFlowDirection IFlowDirectionController.EffectiveFlowDirection
+		{
+			get { return _effectiveFlowDirection; }
+			set
+			{
+				if (value == _effectiveFlowDirection)
+					return;
+
+				_effectiveFlowDirection = value;
+
+				var ve = (Parent as VisualElement);
+				ve?.InvalidateMeasureInternal(InvalidationTrigger.Undefined);
+				OnPropertyChanged(VisualElement.FlowDirectionProperty.PropertyName);
+			}
+		}
+		IFlowDirectionController FlowController => this;
```

Comment:
if I understand this right, you added this property so you don't have to cast to `IFlowDirectionController`everytime you want to invoke an explicitly implemented interface method ? If so, make it a field
```
IFlowDirectionController _flowController = this;
```

Reply:
Can't set a field to `this`.